k9mail-library/src/main/java/com/fsck/k9/mail/store/webdav/WebDavStore.java

```diff
@@ -1042,6 +1040,18 @@ private InputStream sendRequest(String url, String method, StringEntity messageB
         return null;
     }
 
+    private void handleUnexpectedRedirect(HttpResponse response, String url) throws IOException {
+        if (response.getFirstHeader("Location") != null) {
+            // TODO: This may indicate lack of authentication or may alternatively be something we should follow
```

Comment:
In the future please try to write this as instructions (to your future self or others). In this case maybe: "Figure out whether this response indicates lack of authentication or is a redirect we need to expect and follow".